### PR TITLE
Port dependabot dotnet

### PR DESCRIPTION
**Pull request by IEvangelist**
Port over the existing `dependadotnet` written by @richlander as discussed in https://github.com/dotnet/docs/pull/31584#issuecomment-1267373316 and https://github.com/dotnet/docs/pull/31584#issuecomment-1268395368.

- Update this project to .NET 7
- Update the source code a bit from the original
  - Use the `Microsoft.Extensions.FileSystemGlobbing` NuGet 📦.
  - Use `static` lambdas where possible.
  - Use `readonly record struct` where possible.
  - Use source generating `Regex` where possible.
  - Use a `StringBuilder` as a `buffer` and allow for the content to be written a target custom _.github/dependabot.yml_ file path (from the second parameter when running the app):

    ```bash
    dependadotnet run "path/to/docs/directory/" ".github/dependabot.yml"
    ```

    > The destination to write the YML file to defaults to `".github/dependabot.yml"`.

  - Use the local file, rather than a hard-coded raw GitHub source path.
  - Use range expression operators where possible.
  - Use ordering on dir/file names to help keep consistent generated results, limiting the complexity of potentially reviewing random order generation.

This is now exposed as a composite GitHub Action so that we can have various .NET repositories calling into this to automate PRs to keep their @dependabot config files up-to-date.

/cc @nschonni

### CODE REVIEW

File: actions/dependabotdotnet/src/dependabotdotnet/Program.cs [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+StringBuilder buffer = new();
+
+// yaml top-matter
+string topMatter = """
+    # generated by dependadotnet
+    # https://github.com/dotnet/core/tree/main/samples/dependadotnet
+    version: 2
+    updates:
+    """;
```
Comment (nschonni) [inline actions/dependabotdotnet/src/dependabotdotnet/Program.cs:25]: May want to include the actions part too https://github.com/dotnet/core/pull/7845

File: actions/dependabot-bot/src/dependabot-bot/Program.cs [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    }
+}
+
+/*
+Dependabot encountered the following error when parsing your .github/dependabot.yml:
+
+The property '#/updates' had more items than the allowed 200
+Please update the config file to conform with Dependabot's specification. 
+*/
+const int UpdateNodeLimit = 200;
```
Comment (BillWagner) [inline actions/dependabot-bot/src/dependabot-bot/Program.cs:32]: Can you run some tests bumping this up? I haven't tried in a couple of years, and I'm hoping they've removed this restriction. (I looked and the issue I created has been closed. If it's still there, I can reopen it).

File: actions/dependabot-bot/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# dependadotnet -- Generator for dependabot.yml for .NET
```
Comment (BillWagner) [inline actions/dependabot-bot/README.md:1]: You should probably add a section here to describe using it as an action.

File: actions/dependabot-bot/src/dependabot-bot/Program.cs (hunk 2 above)
Comment (IEvangelist) [reply actions/dependabot-bot/src/dependabot-bot/Program.cs:32]: I added this as I literally ran into this limitation this morning, it errored out immediately in the PR while attempting to parse it. I copied and pasted the actual error as a comment about this.

File: actions/dependabot-bot/README.md (hunk 3 above)
Comment (IEvangelist) [reply actions/dependabot-bot/README.md]: Done

**Comment by richlander** [comment]
This is a great idea. I'm only upset I didn't think of it. J/k. Great plan and it is nice seeing this bit of code getting a second life with a much better distribution approach. Great initiative!

File: actions/dependabot-bot/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        default: 'Manual run'
+
+# Run on the latest version of Ubuntu
+jobs:
+  dependabot-bot:
+    runs-on: ubuntu-latest
+
+    # Checkout the repo into the workspace within the VM
+    steps:
+    - uses: actions/checkout@v2
```
Comment (nschonni) [inline actions/dependabot-bot/README.md:72]:
```suggestion
    - uses: actions/checkout@v3
```

**Review by BillWagner — APPROVED** [review]
This LGTM @IEvangelist 

Let's :shipit:

**Pull request merged**